DOTHUB_invertJacobian.m

```diff
@@ -173,16 +175,69 @@
                 S=svd(JJT);
                 invJ{i} = Jtmp'/(JJT + eye(length(JJT))*(varInputs.hyperParameter*max(S)));
             end
-            
         case 'covariance'
-            fprintf('Covariance reconstruction coming soon \n');
-            return
-            
+            fprintf('Under construction...\n');
+            invJ = cell(nWavs,1);
+            for i = 1:nWavs
+                Jtmp = JNatCropped{i};
+                %Extract relevant chunk of prepro.dod to calculate covariance. %Might be better to pass the reference data directly?
+                if length(find(prepro.tDOD<0))>1  %Must be an HRF
+                    [~,ind] = min(abs(prepro.tDOD));
+                    covData = prepro.dod(1:ind,:);
+                else %take first 30 seconds, or 10% of data, whichever is less.
+                    if range(tDOD)< 30
+                        covData = prepro.dod;
+                    else
+                        fs = 1/mean(diff(tDOD));
+                        covData = prepro.dod(1:round(fs*30),:);
+                    end
+                end   
+                sigma_v = cov(covData); 
+                sigma_u = sparse(1:2*nNodeNat,1:2*nNodeNat,1);
+                JJT = Jtmp*sigma_u*Jtmp';
+                l1 = hyperParameter*trace(JJT)/trace(sigma_v);
+                invJ{i} = sigma_u*Jtmp' / ( JJT + sigma_v*l1);
+            end
         case 'spatial'
-            fprintf('Spatially varying regularization reconstruction coming soon \n');
-            return
-            %load rmap
```

Comment:
Don't you need the mesh information for spatially varying reconstruction?  I thought the idea was that lambda is different in the scalp/skull layers?  That's why I had a 'load rmap' comment

Reply:
The second regularization is to bound a column normalization: To avoid placing absorption changes on the surface, the idea is to normalize the sensitivity within each voxel, which can be done by column normalization, but some columns have very low sensitivity (very deep in the brain), so to avoid division by ~zero, there is a second regularization. I guess this can be connected somehow to the mesh, but it might be easier to tune the parameters with something like a 2D L-curve.